cli/lock.ml

```diff
@@ -189,25 +192,78 @@ let display_verbose_diagnostics = function
   | None -> false
   | Some l -> l >= Logs.Info
 
-let interpret_solver_error ~switch_state = function
+let interpret_solver_error ~repositories solver = function
   | `Msg _ as err -> err
   | `Diagnostics d ->
-      (match Opam_solve.not_buildable_with_dune d with
+      (match Opam_solve.not_buildable_with_dune solver d with
       | [] -> ()
       | offending_packages ->
-          check_dune_universe_repo ~switch_state offending_packages);
+          check_dune_universe_repo ~repositories offending_packages);
       let verbose = display_verbose_diagnostics (Logs.level ()) in
-      Opam_solve.diagnostics_message ~verbose d
-
-let calculate_opam ~build_only ~allow_jbuilder ~local_opam_files ~ocaml_version
-    ~target_packages =
+      Opam_solve.diagnostics_message ~verbose solver d
+
+(** Turns each repository URL into a path to the repository's sources,
+    eventually fetching them from the remote. *)
+let make_repository_locally_available url =
+  match OpamUrl.local_dir url with
+  | Some path when Opam.Url.is_local_filesystem url ->
+      Ok (OpamFilename.Dir.to_string OpamFilename.Op.(path / "packages"))
+  | _ ->
+      (* TODO before release *)
```

Comment:
what is this comment about?

Reply:
This is an actual todo, to add support for git URLs